core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -347,72 +413,65 @@ public InnerHitBuilder setHighlightBuilder(HighlightBuilder highlightBuilder) {
         return this;
     }
 
-    public QueryBuilder<?> getQuery() {
+    QueryBuilder<?> getQuery() {
         return query;
     }
 
-    public InnerHitBuilder setQuery(QueryBuilder<?> query) {
+    void setQuery(QueryBuilder<?> query) {
         this.query = Objects.requireNonNull(query);
-        return this;
     }
 
-    public InnerHitBuilder setInnerHitsBuilder(InnerHitsBuilder innerHitsBuilder) {
-        this.innerHitsBuilder = innerHitsBuilder;
-        return this;
+    void setChildInnerHits(Map<String, InnerHitBuilder> childInnerHits) {
+        this.childInnerHits = childInnerHits;
     }
 
-    public InnerHitsContext.BaseInnerHits buildInline(SearchContext parentSearchContext, QueryShardContext context) throws IOException {
-        InnerHitsContext.BaseInnerHits innerHitsContext;
-        if (nestedPath != null) {
-            ObjectMapper nestedObjectMapper = context.getObjectMapper(nestedPath);
-            ObjectMapper parentObjectMapper = context.nestedScope().getObjectMapper();
-            innerHitsContext = new InnerHitsContext.NestedInnerHits(
-                    name, parentSearchContext, parentObjectMapper, nestedObjectMapper
-            );
-        } else if (parentChildType != null) {
-            DocumentMapper documentMapper = context.getMapperService().documentMapper(parentChildType);
-            innerHitsContext = new InnerHitsContext.ParentChildInnerHits(
-                    name, parentSearchContext, context.getMapperService(), documentMapper
-            );
-        } else {
-            throw new IllegalStateException("Neither a nested or parent/child inner hit");
+    void setParentChildType(String parentChildType) {
+        this.parentChildType = parentChildType;
+    }
+
+    String getParentChildType() {
+        return parentChildType;
+    }
+
+    void setNestedPath(String nestedPath) {
+        this.nestedPath = nestedPath;
+    }
```

Comment:
can we get rid of setParentChildType and setNestedPath now that they seem to be set through the constructore above?